lib/src/observable.dart

```diff
@@ -1,92 +1,131 @@
+  StreamController<StreamValueType> /*?*/ controller;
+  Pointer<OBX_observer> /*?*/ _cObserver;
+  ReceivePort /*?*/ receivePort;
+
+  int get nativePort => receivePort /*!*/ .sendPort.nativePort;
+
+  set cObserver(Pointer<OBX_observer> value) {
+    _cObserver = checkObxPtr(value, 'observer initialization failed');
+    _debugLog('started');
   }
 
-  static void subscribe(Store store) {
-    syncOrObserversExclusive.mark(store);
+  Stream<StreamValueType> get stream => controller /*!*/ .stream;
 
-    final callback = Pointer.fromFunction<obx_observer>(_anyCallback);
-    final storePtr = store.ptr;
-    _anyObserver[storePtr.address] =
-        bindings.obx_observe(storePtr, callback, storePtr.cast<Void>());
-    StoreCloseObserver.addListener(store, _anyObserver[storePtr.address], () {
-      unsubscribe(store);
-    });
+  _Observer() {
+    initializeDartAPI();
+  }
+
+  // start() is called whenever user starts listen()-ing to the stream
+  void init(void Function() start) {
+    controller = StreamController<StreamValueType>(
+        onListen: start, onPause: stop, onResume: start, onCancel: stop);
   }
 
-  // #53 ffi:Pointer finalizer
-  static void unsubscribe(Store store) {
-    final storeAddress = store.ptr.address;
-    if (!_anyObserver.containsKey(storeAddress)) {
-      return;
-    }
-    StoreCloseObserver.removeListener(store, _anyObserver[storeAddress]);
-    bindings.obx_observer_close(_anyObserver[storeAddress]);
-    _anyObserver.remove(storeAddress);
-    syncOrObserversExclusive.unmark(store);
+  // stop() is called when the stream subscription is paused or canceled
+  void stop() {
+    _debugLog('stopped');
+    if (_cObserver != null) checkObx(bindings.obx_observer_close(_cObserver));
   }
 
-  static bool isSubscribed(Store store) =>
-      _Observable._anyObserver.containsKey(store.ptr.address);
+  void _debugLog(String message) {
+    // print('Observer=${_cObserver?.address} ' + message);
+  }
 }
 
-extension Streamable<T> on Query<T> {
-  void _setup() {
-    if (!_Observable.isSubscribed(store)) {
-      _Observable.subscribe(store);
-    }
-    final storeAddress = store.ptr.address;
-
-    _Observable._any[storeAddress] ??= <int, Any>{};
-    _Observable._any[storeAddress] /*!*/ [entityId] ??= (u, _, __) {
-      // dummy value to trigger an event
-      _Observable.controller.add(entityId);
-    };
+/// StreamController implementation inspired by the sample controller sample at:
+/// https://dart.dev/articles/libraries/creating-streams#honoring-the-pause-state
+/// https://dart.dev/articles/libraries/code/stream_controller.dart
+extension ObservableStore on Store {
+  /// Create a stream to data changes on EntityT (stored Entity class).
+  ///
+  /// The stream receives an event whenever an object of EntityT is created or
+  /// changed or deleted. Make sure to close() the subscription after you're
+  /// done with it to avoid hanging change listeners.
+  Stream<void> subscribe<EntityT>() {
```

Comment:
I believe you should store `observers` for entity type locally and reuse them.

Reply:
yes, planning to take a look at that

Reply:
Follow up issue #195